### PR TITLE
iidm new import/export modes

### PR DESCRIPTION
**Pull request by ChamseddineBhd**
Linked to powsybl/powsybl-core#608

### CODE REVIEW

File: docs/iidm/importer/iidm.md [hunk 1]
```diff
@@ -12,6 +12,14 @@ For more information about the IIDM model, see [here](../model/index.md).
 IIDM networks can be serialized in XML files. The IIDM importer supports files with the following extensions: `*.xml`,
 `*.xiidm` and `*.iidm`.
 
+The IIDM importer has three importing modes:
+
+- **First mode** : Imports the network and its extensions from a unique file.
+    
+- **Second mode** : Imports the network from a file and the extensions from another file. 
+    
+- **Third mode** : Imports the network from a file and each extension type from a separate file.
```
Comment (MioRtia) [inline docs/iidm/importer/iidm.md]: Same remarks as above about third mode

File: docs/iidm/importer/iidm.md [hunk 2]
```diff
@@ -68,6 +76,29 @@ The `iidm.import.xml.throw-exception-if-extension-not-found` property is an opti
...
+   - `IidmImportExportMode.UNIQUE_FILE`: Imports the network and its extensions from a unique file.
+    
+   - `IidmImportExportMode.EXTENSIONS_IN_ONE_SEPARATED_FILE`: Imports the network from a file and the extensions from another file. 
+      In this case if the network file name is network.xiidm, the extensions file name must be network-ext.xiidm.
+    
+   - `IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE`: Imports the network from a file and each extension type from a separate file.
+      In this mode each extension file name must be networkName-extensionName.xiidm.
+      Example : if we have an extension file for the loadFoo extension type and our network name is test, the file name must be test-loadFoo.xiidm.
+      
+The default value of this parameter is `IidmImportExportMode.NO_SEPARATED_FILE_FOR_EXTENSIONS`.
```
Comment (MioRtia) [inline docs/iidm/importer/iidm.md]: Same remarks as above about `UNIQUE_FILE`

File: docs/iidm/importer/iidm.md [hunk 3]
```diff
@@ -68,6 +76,29 @@ The `iidm.import.xml.throw-exception-if-extension-not-found` property is an opti
...
+Its possible values are :
+
+   - `IidmImportExportMode.UNIQUE_FILE`: Imports the network and its extensions from a unique file.
+    
+   - `IidmImportExportMode.EXTENSIONS_IN_ONE_SEPARATED_FILE`: Imports the network from a file and the extensions from another file. 
+      In this case if the network file name is network.xiidm, the extensions file name must be network-ext.xiidm.
+    
+   - `IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE`: Imports the network from a file and each extension type from a separate file.
+      In this mode each extension file name must be networkName-extensionName.xiidm.
+      Example : if we have an extension file for the loadFoo extension type and our network name is test, the file name must be test-loadFoo.xiidm.
```
Comment (MioRtia) [inline docs/iidm/importer/iidm.md]: *For example, if

Put the names into `` : 
`test loadFoo test-loadFoo.xiidm`

File: docs/iidm/exporter/iidm.md [hunk 4]
```diff
@@ -90,6 +94,28 @@ network. Its default value is `NODE_BREAKER`.
...
+
+   - `IidmImportExportMode.EXTENSIONS_IN_ONE_SEPARATED_FILE`: in case we want to export the network in a file and  the extensions in a separate file.
+
+   - `IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE`: in case we want to export network in a file and each extension type in a separate file. 
+
+The default value for this parameter is `IidmImportExportMode.NO_SEPARATED_FILE_FOR_EXTENSIONS`.
+
+## iidm.export.xml.extensions
+The `iidm.export.xml.extensions` property is an optional property that defines the list of extensions that we want to export by the XIIDM exporter. 
+by default all extensions will be exported.
```
Comment (MioRtia) [inline docs/iidm/exporter/iidm.md]: *By

File: docs/iidm/importer/iidm.md [hunk 5]
```diff
@@ -68,6 +76,29 @@ The `iidm.import.xml.throw-exception-if-extension-not-found` property is an opti
...
+      
+The default value of this parameter is `IidmImportExportMode.NO_SEPARATED_FILE_FOR_EXTENSIONS`.
+
+## iidm.import.xml.extensions
+The `iidm.import.xml.extensions` property is an optional property that defines the list of extensions that we want to import by the XIIDM importer. 
+by default all extensions will be imported.
```
Comment (MioRtia) [inline docs/iidm/importer/iidm.md]: *By

File: docs/iidm/exporter/iidm.md [hunk 6]
```diff
@@ -90,6 +94,28 @@ network. Its default value is `NODE_BREAKER`.
...
+The `iidm.export.xml.export-mode` property is an optional property that defines the export mode of the XIIDM exporter.
+the export mode can be:
+
+   - `IidmImportExportMode.UNIQUE_FILE`: in case we want to export the network and its extensions in a unique file, 
+
+   - `IidmImportExportMode.EXTENSIONS_IN_ONE_SEPARATED_FILE`: in case we want to export the network in a file and  the extensions in a separate file.
+
+   - `IidmImportExportMode.ONE_SEPARATED_FILE_PER_EXTENSION_TYPE`: in case we want to export network in a file and each extension type in a separate file. 
+
+The default value for this parameter is `IidmImportExportMode.NO_SEPARATED_FILE_FOR_EXTENSIONS`.
```
[the remark this answers is not in the document]
Comment (ChamseddineBhd) [reply docs/iidm/exporter/iidm.md]: This was pushed before changing the first enum value.

File: docs/iidm/exporter/iidm.md [hunk 7]
```diff
@@ -11,6 +11,14 @@ For more information about the IIDM model, see [here](../model/index.md).
 
 IIDM networks can be serialized in XML files. The IIDM exporter generates files with a `*.xiidm` extension.
 
+The IIDM exporter has three exporting modes:
+
+   - **First mode**: Exports the network and its extensions in a unique file.
+    
+   - **Second mode**: Exports the network in a file and the extensions in another file.
+    
+   - **Third mode**: Exports the network in a file and each extension type in a separate file.
```
[the remark this answers is not in the document]
Comment (ChamseddineBhd) [reply docs/iidm/exporter/iidm.md]: i have added an example in the param details section. Here we just have to  inform the reader that there are three export mode.

File: docs/iidm/importer/iidm.md (hunk 1 above)
Comment (ChamseddineBhd) [reply docs/iidm/importer/iidm.md]: Same response

File: docs/iidm/importer/iidm.md [hunk 8]
```diff
@@ -68,10 +76,32 @@ The `iidm.import.xml.throw-exception-if-extension-not-found` property is an opti
...
+
+## iidm.import.xml.extensions
+The `iidm.import.xml.extensions` property is an optional property that defines the list of extensions that we want to import by the XIIDM importer. 
+By default all extensions will be imported.
+
 # Deprecated configuration properties for IIDM-XML importer
 
 ## throwExceptionIfExtensionNotFound
-The `throwExceptionIfExtensionNotFound` property is deprecated since v2.0.0. Use the `iidm.import.xml.throw-exception-if-extension-not-found` property instead.
+The `throwExceptionIfExtensionNotFound` property is deprecated since v2..0. Use the `iidm.import.xml.throw-exception-if-extension-not-found` property instead.
```
Comment (MioRtia) [inline docs/iidm/importer/iidm.md]: mistake ? why did you change the version since this property is deprecated
